zio-config/src/main/scala/zio/config/package.scala

```diff
@@ -27,4 +27,19 @@ package object config extends Sources {
   def report[A](config: => Config[A]): Report[A] = Report.report[A](config)
 
   def getConfigValue(path: String): ZIO[ConfigSource, Unit, String] = ZIO.accessM(_.configService.getConfigValue(path))
+
+  ////
+
+  type NonEmptyList[A] = ::[A]
+
+  object NonEmptyList {
+    def apply[A](a: A, tail: A*): NonEmptyList[A] =
+      ::(a, tail.toList)
+
+    def concat[A](lerr: NonEmptyList[A], rerr: NonEmptyList[A]): NonEmptyList[A] =
+      ::(lerr.head, lerr.tail ++ rerr)
+  }
+
```

Comment:
Idea here is to have NEL reusable in the future.

Reply:
Let's be specific about ReadErrors,